Metrics/start-windows-build.ps1

```diff
@@ -0,0 +1,365 @@
+    $command = $ProcessPath -replace '\\', '\\'
+    if($ArgumentList.Count) {
+        $ArgumentList | Foreach-Object { $command += " $($_ -replace '\\', '\\')" }
+    }
+    try {
+        Wait-ProcessToFinish -ProcessPath $ProcessPath -ArgumentList $ArgumentList `
+                             -StandardOutput $stdoutFile -StandardError $stderrFile
+        $msg = "Successfully executed: $command"
+    } catch {
+        $msg = "Failed command: $command"
+        $global:PARAMETERS["BUILD_STATUS"] = 'FAIL'
+        $global:PARAMETERS["LOGS_URLS"] += $($stdoutUrl, $stderrUrl)
+        $global:PARAMETERS["FAILED_COMMAND"] = $command
+        Write-Output "Exception: $($_.ToString())"
+        Throw $BuildErrorMessage
+    } finally {
+        Write-Output $msg
+        Write-Output "Stdout log available at: $stdoutUrl"
+        Write-Output "Stderr log available at: $stderrUrl"
+    }
+}
+
+function Get-LatestCommitID {
+    if(!$global:LATEST_COMMIT_ID) {
+        Throw "Failed to get the latest dcos-metrics commit ID. Perhaps it has not saved."
+    }
+    return $global:LATEST_COMMIT_ID
+}
+
+function Set-LatestMetricsCommit {
+    Push-Location $METRICS_GIT_REPO_DIR
+    try {
+        if($CommitID) {
+            Start-ExternalCommand { git.exe reset --hard $CommitID } -ErrorMessage "Failed to set Metrics git repo last commit to: $CommitID."
+        }
+        Start-ExternalCommand { git.exe log -n 1 } -ErrorMessage "Failed to get the latest commit message for the Metrics git repo" | Out-File "$METRICS_BUILD_LOGS_DIR\latest-commit.log"
+        $commitId = Start-ExternalCommand { git.exe log --format="%H" -n 1 } -ErrorMessage "Failed to get the latest commit id for the dcos-metrics git repo"
+        Set-Variable -Name "LATEST_COMMIT_ID" -Value $commitId -Scope Global
+    } finally {
+        Pop-Location
+    }
+}
+
+function New-TestingEnvironment {
+    Write-Output "Creating new tests environment"
+    Start-EnvironmentCleanup # Do an environment cleanup to make sure everything is fresh
+    New-Directory $METRICS_DIR
+    New-Directory $METRICS_BUILD_OUT_DIR -RemoveExisting
+    New-Directory $METRICS_BUILD_LOGS_DIR
+    $global:PARAMETERS["BRANCH"] = $Branch
+    Start-GitClone -Path $METRICS_GIT_REPO_DIR -URL $GitURL -Branch $Branch
+    Set-LatestMetricsCommit
+    Start-GitClone -Path $METRICS_DCOS_WINDOWS_GIT_REPO_DIR -URL $DCOS_WINDOWS_GIT_URL
+    Start-ExternalCommand { git.exe config --global user.email "ostcauto@microsoft.com" } -ErrorMessage "Failed to set git user email"
+    Start-ExternalCommand { git.exe config --global user.name "ostcauto" } -ErrorMessage "Failed to set git user name"
+    Write-Output "New tests environment was successfully created"
+}
+
+function Start-DCOSMetricsBuild {
+    Write-Output "Building DC/OS Metrics in $METRICS_GIT_REPO_DIR"
+    Push-Location $METRICS_GIT_REPO_DIR
+    try {
+        New-Item -ItemType directory -Path ".\build" -Force
+        & $GOLANG_DIR\bin\go get .\...
```

Comment:
Please wrap this with the `Start-ExternalCommand`. The whole script won't fail if this exits with code != 0.

The `Start-ExternalCommand` is just a [simple wrapper](https://github.com/Microsoft/mesos-jenkins/blob/master/Modules/CIUtils/CIUtils.psm1#L42) from our `CIUtils` module that does this checking for you. 
See usage example [here](https://github.com/Microsoft/mesos-jenkins/blob/master/Mesos/start-windows-build.ps1#L186).

Reply:
Advice taken.